build-system/compile/compile-utils.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Since we no longer pass the process_common_js_modules flag to closure
+// compiler, we must now tranform these common JS node_modules to ESM before
+// passing them to closure.
+// TODO(rsimha, erwinmombay): Derive this list programmatically if possible.
+const commonJsModules = [
+  'node_modules/dompurify/',
+  'node_modules/promise-pjs/',
+  'node_modules/set-dom/',
```

Comment:
@rsimha added RRule, too.

Reply:
Only those `dependencies` in `package.json` that are `commonJs` modules need to be in this list. If the build works without adding `rrule` to this, we should leave it as is.

Reply:
This is going away in #23106